src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/ChunkContextUi.java

```diff
@@ -152,6 +167,31 @@ public void execute()
             "Don't Run", true);
    }
 
+   @Override
+   public void switchChunk(String chunkType)
+   {
+      if (chunk_ != null)
+      {
+         DocDisplay docDisplay = target_.getDocDisplay();
+         
+         Position start = chunk_.getPreamble();
+         Position end = chunk_.getEnd();
+         
+         String chunkText = docDisplay.getTextForRange(Range.fromPoints(start, end));
+         JsArrayString chunkLines = StringUtil.split(chunkText, "\n");
+         if (chunkLines.length() > 0)
+         {
+            String firstLine = chunkLines.get(0);
+            Position linedEnd = Position.create(start.getRow(),firstLine.length());
+            
+            String newFirstLine = firstLine.replaceFirst("{[a-zA-Z]+", "{" + chunkType);
```

Comment:
Does/should this support specifying the engine as an explicit chunk option? e.g. ```{r, engine='python'} ?